### PR TITLE
Alternative Idle Handling

### PR DESCRIPTION
**Pull request by helloworld111gh**
Some people complained that they could be ready but considered "idle" and keep having the extension turn off on them. This PR will fix it.

Instead of just turning off after a delay, the idle handing will instead prompt the user to accept any HIT's they get if they are considered idle. If they don't accept/do nothing (2 chances to accept) the extension turns off.

Also the extension will turn off if the computer is locked or screensaver.

Note: This will break if HIT reservation is added, as it relies on HIT's not being reserved until the tab opens

[This PR was earlier closed because there was a bug, but after browser restart I couldn't reproduce it.]

### CODE REVIEW

**Comment by ukd1** [comment]
@helloworld111gh I'm reviewing, but if it does what you say it's bad-ass!

BTW, we're talking internally about adding reservation very soon...will be good for everyone!

File: extension/js/main.js [hunk 1]
```diff
@@ -192,25 +195,53 @@ chrome.identity.getProfileUserInfo(function(info) {
...
-    shut_off_timer = setTimeout(function() {
-      if (info_hash.tester_state === 'idle') {
-        _checking_active = false;
-        set_checking(_checking_active);        
-      }
-    },default_check_for_work_interval * 45);
   } else if (state === 'active') {
-    clearTimeout(shut_off_timer);
     check_for_work_interval = default_check_for_work_interval;
+  } else if (state === 'locked') {
```
Comment (ukd1) [inline extension/js/main.js]: nice find

**Comment by ukd1** [comment]
@helloworld111gh ok, this looks good - I need to test it out here before we ship this - I don't think I can do before Monday though. Thanks so much for this.

BTW, we're hiring developers if you like this kinda thing?

**Comment by mkdige** [comment]
"Also the extension will turn off if the computer is locked or screensaver." 
Not quite, extension goes off if you don't do anything on chrome (not moving mouse, open/move tabs, etc) for a certain period of time.
In any case if chrome remains unchanged for a time period the extension goes off.
I am not sure is this was the idea, but nevertheless longer period of inactivity (say ~30 minutes) would be nicer

**Comment by ukd1** [comment]
@mkdige this is a new version that's not released. The current version turns off after 6 minutes.

**Comment by mkdige** [comment]
Got it, sorry misunderstanding.
BRIgor

**Comment by warpten09025** [comment]
Please change from 6 minutes to 15 minutes ! That would be nicer

**Comment by helloworld111gh** [comment]
Should I update this to match current codebase?

**Comment by helloworld111gh** [comment]
gonna update this to 1.2.x with new check to prevent auto-refresh when idle - will create new PR when this is completed

**Pull request closed without merging**